administrator/components/com_joomlaupdate/tmpl/update/default.php

```diff
@@ -21,18 +24,12 @@
 $ajaxUrl = JUri::base() . 'components/com_joomlaupdate/restore.php';
 $returnUrl = 'index.php?option=com_joomlaupdate&task=update.finalise&' . JFactory::getSession()->getFormToken() . '=1';
 
-JFactory::getDocument()->addScriptDeclaration(
-	"
-	var joomlaupdate_password = '$password';
-	var joomlaupdate_totalsize = '$filesize';
-	var joomlaupdate_ajax_url = '$ajaxUrl';
-	var joomlaupdate_return_url = '$returnUrl';
-
-	jQuery(document).ready(function(){
-		window.pingExtract();
-		});
-	"
-);
+HTMLHelper::_('script', 'com_joomlaupdate/admin-update-default.js', ['relative' => true, 'version' => 'auto']);
+
+Factory::getDocument()->addScriptOptions('joomlaupdate_password', $password);
```

Comment:
This needs to be to an array of options rather than calling the method 4 times.

```
Factory::getDocument()->addScriptOptions(
	'joomlaupdate',
	[
                'joomlaupdate_password' => $password,
                'joomlaupdate_totalsize' => $filesize,
                'joomlaupdate_ajax_url' => $ajaxUrl,
                'joomlaupdate_return_url' => $returnUrl,
	]
);
```